source/javascripts/lib/wirsing/utils/pager.coffee

```diff
@@ -0,0 +1,14 @@
+#= require ../module
+
+# creates a list that makes paginating a list simple
+
+wirsing.module 'utils.Pager', ->
+  class Pager
+    constructor: (@options={}) ->
+      @pageSize     = @options.perPage || 30
+      @list         = []
+      @visibleCount = @pageSize
+
+    setList: (newList) -> @list = newList
+    nextPage:    -> @visibleCount += @pageSize
+    currentList: -> @list.slice(0,@visibleCount)
```

Comment:
Because it took me a half hour to even understand how you were creating the filters. I'm not used to JavaScript as Haskell.